### PR TITLE
Makes CustomTabsServiceConnection class static | Stress testing

### PR DESCRIPTION
**Pull request by iambmelt**
This change converts the `CustomTabsServiceConnection` to a `static class` and removes strong references to its enclosing `class`

### CODE REVIEW

File: msal/src/main/java/com/microsoft/identity/client/AuthenticationActivity.java [hunk 1]
```diff
@@ -130,22 +131,38 @@ private void warmUpCustomTabs() {
...
-    @NonNull
-    private CustomTabsServiceConnection createCustomTabsServiceConnection() {
-        return new CustomTabsServiceConnection() {
-            @Override
-            public void onCustomTabsServiceConnected(ComponentName name, CustomTabsClient client) {
-                mCustomTabsServiceIsBound = true;
-                mCustomTabsClient = client;
-                mCustomTabsClient.warmup(0L);
-                mCustomTabsSession = mCustomTabsClient.newSession(null);
+    private static class MsalCustomTabsServiceConnection extends CustomTabsServiceConnection {
```
Comment (weijjia) [inline msal/src/main/java/com/microsoft/identity/client/AuthenticationActivity.java]: I'm a little bit worried about the implementation 1) connection class having reference to enclosing class doesn't look good :) 2) in line 121, we call bindCustomTabsService, which is an asynchronous call, but we pass in the customtab session right after it to intent builder, is it possible that it's not connected yet? 3) I don't think we're using warm up at all. If you look at the onResume, it's creating a new customTabIntent and that is the intent used to launch URL.

**Comment by iambmelt** [comment]
@weijjia f4289310936c9d840b90f19f0f553cf9ea8f71b6 shows what this `Activity` would look like if we could skip `warmup()`

**Comment by weijjia** [comment]
so much cleaner :)

File: msal/src/main/java/com/microsoft/identity/client/AuthenticationActivity.java [hunk 2]
```diff
@@ -125,29 +131,70 @@ private void warmUpCustomTabs() {
...
+
+        private final WeakReference<AuthenticationActivity> mAuthenticationActivityWeakReference;
```
Comment (weijjia) [inline msal/src/main/java/com/microsoft/identity/client/AuthenticationActivity.java]: Like I mentioned above, having a reference to enclosing class is a bad sign. If it won't cause too much trouble, i would suggest we have the customsTabSession and customTabsClient declared inside of the connection, once it's connected you'll set those values. We can do connection.getTabSesson or connection.getTabsClient for it's enclosing class to use.

**Pull request merged**